app/src/main/java/org/p2p/wallet/striga/StrigaUserIdProvider.kt

```diff
@@ -1,9 +1,12 @@
 package org.p2p.wallet.striga
 
-import org.p2p.wallet.utils.emptyString
+import org.p2p.wallet.auth.interactor.MetadataInteractor
 
-class StrigaUserIdProvider {
+class StrigaUserIdProvider(
+    private val metadataInteractor: MetadataInteractor
+) {
     fun getUserId(): String {
-        return emptyString()
+        return metadataInteractor.currentMetadata?.strigaMetadata?.userId
+            ?: error("Striga userId has not been set to metadata")
```

Comment:
I suggest returning just null and handle case if it's null in the place where this method was called